gnovm/pkg/gnolang/nodes.go

```diff
@@ -1333,16 +1333,23 @@ func NewPackageNode(name Name, path string, fset *FileSet) *PackageNode {
 	return pn
 }
 
-func (pn *PackageNode) NewPackage() *PackageValue {
-	pv := &PackageValue{
-		Block: &Block{
-			Source: pn,
-		},
-		PkgName:    pn.PkgName,
-		PkgPath:    pn.PkgPath,
-		FNames:     nil,
-		FBlocks:    nil,
-		fBlocksMap: make(map[string]*Block),
+func (pn *PackageNode) NewPackage(alloc *Allocator) *PackageValue {
+	var pv *PackageValue
+	if pn.PkgName == "main" {
+		// Allocation is only for the new created main package,
+		// other packages are allocted while loading from store.
```

Comment:
Where is package value allocation counted when loading from the store? Can you point it out?
This is the only place where AllocPackageValue() is called, through alloc.NewPackageValue(pn).

Reply:
The main package is built for a file test, a MsgCall (see vm/keeper.go), or MsgRun. It is not stored, but it will still be scanned during GC.

Reply:
and for other packages, see loadObjectFromStore.